gnomad_qc/v4/annotations/generate_freq_genomes.py

```diff
@@ -1366,35 +1423,80 @@ def main(args):
             ht.count(),
         )
 
-        logger.info(
-            "Annotating call stats group membership for each v3.1 release sample..."
-        )
         v3_sites_meta_ht = v3_vds.variant_data.cols()
         v3_sites_meta_ht = v3_sites_meta_ht.filter(v3_sites_meta_ht.meta.release)
-        group_membership_ht = get_group_membership_ht_for_an(v3_sites_meta_ht)
-        group_membership_ht = group_membership_ht.checkpoint(
-            new_temp_file("group_membership_all", "ht")
-        )
 
         logger.info(
-            "Computing the AN HT of v3.1 samples for new v4.0 genomes variants..."
+            "Annotating call stats group membership for pop diff release samples..."
+        )
+        # Need to make sure we are using the names that are in the vds for the release
+        # samples.
+        # Note: the samples in the pop_diff HT are samples in the to-be-split 'Han' and
+        # 'Papuan' populations AND their 'gnomad_release' status hasn't changed.
+        v3_sites_meta_rename_ht = v3_sites_meta_ht.key_by(
+            s_no_prefix=v3_sites_meta_ht.s.replace("v3.1::", "")
+        )
+        pop_diff_sample_ht = get_updated_release_samples(res.updated_meta_ht.ht())[0]
+        pop_diff_sample_ht = pop_diff_sample_ht.key_by(
+            s=v3_sites_meta_rename_ht[pop_diff_sample_ht.s].s
+        )
+        pop_diff_group_membership_ht = get_group_membership_ht_for_an(
+            pop_diff_sample_ht, only_pop_diff=True
+        )
+        pop_diff_group_membership_ht = pop_diff_group_membership_ht.checkpoint(
+            new_temp_file("group_membership_pop_diff", "ht")
         )
-        ht = compute_an_by_group_membership(v3_vds, group_membership_ht, ht)
-        ht.write(res.v3_release_an_ht.path, overwrite=overwrite)
+        logger.info(
+            "Computing the AN HT of v3.1 pop diff samples for all v4.0 genomes"
+            " variants. Freq meta: %s",
+            hl.eval(pop_diff_group_membership_ht.freq_meta),
+        )
+        ht = compute_an_by_group_membership(
+            v3_vds, pop_diff_group_membership_ht, res.freq_join_ht.ht().select()
+        )
```

Comment:
```suggestion
        # To avoid double counting, we need to filter out the variants that are in
        # v4.0 genomes but not already present in pop_diff samples to get the AN.
        join_ht = res.freq_join_ht.ht()
        join_ht = join_ht.filter(~hl.is_defined(join_ht.ann_array.freq[1])).select()
        ht = compute_an_by_group_membership(
            v3_vds, pop_diff_group_membership_ht, join_ht
        )
```

Reply:
shouldn't it be `join_ht.ann_array[1].freq`?

Reply:
It seems to get the same thing, but yours seems to be accurate.

Reply:
interesting, I wouldn't have expected that to give the same thing

Reply:
can I change it to `hl.is_missing(join_ht.ann_array[1].freq)`? and then do you think it's ready?

Reply:
Yeah, I changed it for you.

Reply:
thank you!